pkg/restore/restore.go

```diff
@@ -979,16 +983,29 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 	name := obj.GetName()
 
 	// Check if we've already restored this itemKey.
-	itemKey := velero.ResourceIdentifier{
-		GroupResource: groupResource,
-		Namespace:     namespace,
-		Name:          name,
+	itemKey := itemKey{
+		resource:  resourceKey(obj),
+		namespace: namespace,
+		name:      name,
 	}
 	if _, exists := ctx.restoredItems[itemKey]; exists {
 		ctx.log.Infof("Skipping %s because it's already been restored.", resourceID)
 		return warnings, errs
 	}
-	ctx.restoredItems[itemKey] = struct{}{}
+	ctx.restoredItems[itemKey] = ""
+	defer func() {
+		// the action field is set explicitly
+		if action := ctx.restoredItems[itemKey]; len(action) > 0 {
+			return
+		}
+		// no action specified, and no warnings and errors
+		if errs.IsEmpty() && warnings.IsEmpty() {
```

Comment:
In which case will this happen? No errors nor warn, but the result was not updated.
Can we deduce the item is skipped for restore for sure when this happens?

Reply:
There are several cases in which the resources are skipped, e.g.
- [The resources are excluded by users](https://github.com/vmware-tanzu/velero/blob/main/pkg/restore/restore.go#L912-L964)
- [The resource is completed](https://github.com/vmware-tanzu/velero/blob/main/pkg/restore/restore.go#L975)
- [Mirror pod](https://github.com/vmware-tanzu/velero/blob/main/pkg/restore/restore.go#L996)
- [PV that backed up by PVB](https://github.com/vmware-tanzu/velero/blob/main/pkg/restore/restore.go#L1089)
- [PV that has delete reclaim policy](https://github.com/vmware-tanzu/velero/blob/main/pkg/restore/restore.go#L1097)
- [Excluded by plugin](https://github.com/vmware-tanzu/velero/blob/main/pkg/restore/restore.go#L1151)
- [The resource already exists and is the same as with backup](https://github.com/vmware-tanzu/velero/blob/main/pkg/restore/restore.go#L1362)

The codes are in different places and are quite easy to miss out. So here I just specify the action for creation and update, and set the action as skip for others that have no errors and warnings.